lhotse/recipes/mgb2.py

```diff
@@ -0,0 +1,202 @@
+QCRI Arabic ASR system has been used to recognize all programs. The ASR output was used to align the manual captioning and produce speech segments for training speech recognition.
+More than 20 hours from 2015 programs have been transcribed verbatim and manually segmented. This data is split into a development set of 10 hours,
+and a similar evaluation set of 10 hours. 
+Both the development and evaluation data have been released in the 2016 MGB challenge
+"""
+from re import sub, match
+from os import path, system
+from string import punctuation
+from logging import info
+from itertools import chain
+from pathlib import Path
+from typing import Dict, Optional, Union
+from lhotse.kaldi import load_kaldi_data_dir
+from lhotse import fix_manifests, validate_recordings_and_supervisions
+from lhotse import Recording, RecordingSet, SupervisionSegment, SupervisionSet
+from lhotse.recipes.utils import manifests_exist, read_manifests_if_cached
+from lhotse.utils import Pathlike, recursion_limit, check_and_rglob, is_module_available
+from itertools import chain
+from bs4 import BeautifulSoup
+
+
+def download_mgb2(
+    target_dir: Pathlike = ".",
+) -> None:
+    """
+    Download and untar the dataset.
+
+    NOTE: This function just returns with a message since MGB2 is not available
+    for direct download.
+
+    :param target_dir: Pathlike, the path of the dir to storage the dataset.
+    """
+    info(
+        "MGB2 is not available for direct download. Please fill out the form at"
+        " https://arabicspeech.org/mgb2 to download the corpus."
+    )
+
+
+def prepare_mgb2(
+    corpus_dir: Pathlike,
+    output_dir: Pathlike,
+    text_cleaning: bool = True,
+    buck_walter: bool = False,
+    num_jobs: int = 1,
+    mer_thresh:int = 80,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+
+    """
+    Returns the manifests which consist of the Recordings and Supervisions.
+    When all the manifests are available in the ``output_dir``, it will simply read and return them.
+
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :param text_cleaning: Bool, if True, basic text cleaning is performed (similar to ESPNet recipe).
+    :param buck_walter: Bool, use BuckWalter transliteration 
+    :param num_jobs: int, the number of jobs to use for parallel processing.
+    :param mer_thresh: int, filter out segments based on mer (Match Error Rate)
+    :return: a Dict whose key is the dataset part, and the value is Dicts with the keys 'audio' and 'supervisions'.
+
+    .. note::
+        Unlike other recipes, output_dir is not Optional here because we write the manifests
+        to the output directory while processing to avoid OOM issues, since it is a large dataset.
+
+    .. caution::
+        The `text_cleaning` option removes all punctuation and diacritics.
+    """
+    if not is_module_available("bs4"):
```

Comment:
Is it too late to check the existence of `bs4` here?

If it is not available, it will fail when you use `bs4` at the beginning of the file
```
from bs4 import BeautifulSoup
```

Reply:
Done

Reply:
I added the import after the condition